ir/ir.rs

```diff
@@ -471,12 +468,39 @@ impl Item {
     }
 
     /// Get this item's name.
-    #[inline]
     pub fn name(&self) -> &str {
-        if let ItemKind::Code(ref code) = self.kind {
-            code.demangled().unwrap_or(&self.name)
-        } else {
-            &self.name
+        match &self.kind {
```

Comment:
Another note, while we're in review: I _sort of_ detest that this is called the item's "kind" if it now also stores the name due to this approach. Would this be best referred to as `ItemMetadata` or something along those lines?

Reply:
heh, we used `kind` in `rust-bindgen`, so it is familiar to me, but I don't have much of a first-principles argument here